src/library/Waveform.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * This Waveform class implements a Waveform UI. To initialize
+ * it, you need to provide the ID of the canvas element where the waveform 
+ * will be rendered, and the fftSize value for the analyserNode. 
+ * The fftSize value defines the window size in samples that is used 
+ * when performing a Fast Fourier Transform (FFT) to get frequency
+ * domain data in the AnalyserNode. When the waveform is created, it requires
+ * an AnalyserNode for real-time audio input data. This class only requires
+ * a canvas element, and there is no additional dependency.
+ */
+
+class Waveform {
+
+  /**
+   * @constructor Constructs a Waveform object for rendering waveform visuals.
+   *   The width and height of the waveform canvas are initialized in the 
+   *   constructor and cannot be changed after initialization.
+   * @param {String} canvasId The ID of the canvas element where the waveform
+   *   will be rendered.
+   * @param {AnalyserNode} analyserNode The AnalyserNode used for visualization.
+   * @param {Number} fftSize The window size in samples used for Fast Fourier 
+   *   Transform (FFT) to obtain frequency domain data in the AnalyserNode.
+   */
+  constructor(canvasId, analyserNode, fftSize) {
+    /** @private @const {!HTMLCanvasElement} Selected waveform canvas element */
+    this.canvas_ = document.querySelector(canvasId);
+    /** @private @const {!CanvasRenderingContext2D} Canvas context */
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    /** @private @const {!number} Selected waveform canvas width */
+    this.width_ = this.canvas_.width;
+    /** @private @const {!number} Selected waveform canvas height */
+    this.height_ = this.canvas_.height;
+    /** @private @const {!number} Current X axis in the waveform */
+    this.currentX_ = 0;
+    /** @private @const {!number} Previous Y axis in the waveform */
+    this.previousY_ = this.height_ / 2;
+    /** @private @const {!AnalyserNode} AnalyserNode of the audio context */
+    this.analyser_ = analyserNode;
+    this.analyser_.fftSize = fftSize;
+    /** @private @const {!number} Total number of data points available at the 
+     * AudioContext sample rate */
+    this.bufferLength_ = this.analyser_.frequencyBinCount;
+    /** @private @const {!Float32Array} AnalyserNode of the audio context */
+    this.dataArray_ = new Float32Array(this.bufferLength_);
+  }
+  
+
+  /**
+   * Render the Waveform on the canvas.
```

Comment:
Please use "ReWrap" VS Code extension for all the comments to wrap at column 80.